### PR TITLE
Add docusaurus scaffold

### PR DESCRIPTION
**Pull request by madebyrogal**
## Description
- I've added docusaurus' scaffold in this PR
- and removed hugo related files

I would treat this PR as a initial commit, 99% of the code is autogenerated by docusaurus.
You can focus only on the files that I've removed/changed

## Related issue(s)
https://github.com/kubeshop/botkube-docs/issues/80

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v16.14
```
Comment (pkosiec) [inline .nvmrc:1]: Let's use the latest Node.js LTS - (16.16) 🙂

File: docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  url: 'https://botkube.io/',
+  baseUrl: '/',
+  onBrokenLinks: 'throw',
+  onBrokenMarkdownLinks: 'warn',
+  favicon: 'images/favicon.png',
+
+  // GitHub pages deployment config.
+  // If you aren't using GitHub pages, you don't need these.
+  organizationName: 'KubeShop', // Usually your GitHub org/user name.
+  projectName: 'botkube', // Usually your repo name.
```
Comment (pkosiec) [inline docusaurus.config.js:20]: We'll use Cloudflare Pages anyway, but we can keep them. Let's remove comments though:

```suggestion
  organizationName: 'kubeshop",
  projectName: 'botkube',
```

File: docusaurus.config.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            href: 'https://github.com/kubeshop/botkube',
+            label: 'GitHub',
+            position: 'right',
+          },
+        ],
+      },
+      footer: {
+        style: 'dark',
+        links: [],
+        copyright: `Copyright © ${new Date().getFullYear()} KubeShop, Inc. Built with Docusaurus.`,
```
Comment (pkosiec) [inline docusaurus.config.js:69]: `S` in `Kubeshop` is lowercase:
```suggestion
        copyright: `Copyright © ${new Date().getFullYear()} Kubeshop, Inc. Built with Docusaurus.`,
```

File: docs/installation.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (pkosiec) [inline docs/installation.md:1]: If we have this document already, then you can put "Documentation" link in the navbar, which points to it 🙂

File: docusaurus.config.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
+// @ts-check
+// Note: type annotations allow type checking and IDEs autocompletion
+
+const lightCodeTheme = require('prism-react-renderer/themes/github');
+const darkCodeTheme = require('prism-react-renderer/themes/dracula');
+
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+  title: 'BotKube :: Messaging bot for monitoring and debugging Kubernetes clusters',
+  tagline: 'BotKube is cool',
```
Comment (pkosiec) [inline docusaurus.config.js:10]:
```suggestion
  title: 'BotKube',
  tagline: 'Messaging bot for monitoring and debugging Kubernetes clusters',
```

File: docusaurus.config.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+  presets: [
+    [
+      'classic',
+      /** @type {import('@docusaurus/preset-classic').Options} */
+      ({
+        docs: {
+          sidebarPath: require.resolve('./sidebars.js'),
+          // Please change this to your repo.
+          // Remove this to remove the "edit this page" links.
```
Comment (pkosiec) [inline docusaurus.config.js:38]:
```suggestion
```

File: docusaurus.config.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  favicon: 'images/favicon.png',
+
+  // GitHub pages deployment config.
+  // If you aren't using GitHub pages, you don't need these.
+  organizationName: 'KubeShop', // Usually your GitHub org/user name.
+  projectName: 'botkube', // Usually your repo name.
+
+  // Even if you don't use internalization, you can use this field to set useful
+  // metadata like html lang. For example, if your site is Chinese, you may want
+  // to replace "en" with "zh-Hans".
```
Comment (pkosiec) [inline docusaurus.config.js:24]:
```suggestion
```

**Review by pkosiec — APPROVED** [review]
Looks good to me 👍 Please apply a few minor commits before merge 🙂

File: docusaurus.config.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+            'https://github.com/kubeshop/botkube-docs',
+        },
+        theme: {
+          customCss: require.resolve('./src/css/custom.css'),
+        },
+      }),
+    ],
+  ],
+
+  themeConfig:
```
Comment (pkosiec) [inline docusaurus.config.js:49]: BTW consider using `colorMode.respectPrefersColorScheme`: https://docusaurus.io/docs/api/themes/configuration#respectPrefersColorScheme

File: docs/installation.md (hunk 4 above)
Comment (madebyrogal) [reply docs/installation.md:1]: Will be in next PR

**Pull request merged**